Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -483,6 +483,108 @@ public void Generate_InvalidSqaleFileSpecified_GeneratorError()
             logger.AssertSingleErrorExists("invalidSqale.xml"); // expecting an error containing the invalid sqale file name
         }
 
+        [TestMethod]
+        public void Generate_RulesFileNotSpecified_TemplateFileCreated()
+        {
+            // Arrange
+            string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+
+            var logger = new TestLogger();
+            var remoteRepoBuilder = new RemoteRepoBuilder(this.TestContext);
+            var child1 = CreatePackageWithAnalyzer(remoteRepoBuilder, "child1.requiredAccept.id", "2.1", License.NotRequired);
+            var child2 = CreatePackageWithAnalyzer(remoteRepoBuilder, "child2.id", "2.2", License.NotRequired);
+            var parent = CreatePackageWithAnalyzer(remoteRepoBuilder, "parent.id", "1.0", License.NotRequired, child1, child2);
+
+            var nuGetHandler = new NuGetPackageHandler(remoteRepoBuilder.FakeRemoteRepo, GetLocalNuGetDownloadDir(), logger);
+
+            var testSubject = new AnalyzerPluginGenerator(nuGetHandler, logger);
+
+            // 1. Generate a plugin for the target package only. Expecting a plugin and a template rule file.
+            var args = new ProcessedArgsBuilder("parent.id", outputDir)
+                .SetLanguage("cs")
+                .SetPackageVersion("1.0")
+                .SetRecurseDependencies(true)
+                .Build();
+            bool result = testSubject.Generate(args);
+
+            Assert.IsTrue(result, "Expecting generation to have succeeded");
+            AssertRuleTemplateFileExistsForPackage(logger, outputDir, parent);
+
+            // 2. Generate a plugin for target package and all dependencies. Expecting three plugins and associated rule files.
+            logger.Reset();
+            args = CreateArgs("parent.id", "1.0", "cs", null, false, true /* /recurse = true */, outputDir);
+            result = testSubject.Generate(args);
+
+            Assert.IsTrue(result, "Expecting generation to have succeeded");
```

Comment:
Use fluent assertions instead (e.g. `testSubject.Generate(args).Should().BeTrue()`).